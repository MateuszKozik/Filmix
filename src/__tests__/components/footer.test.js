import { render } from "@testing-library/react";
import { Footer } from "../../components";

describe("<Footer />", () => {
	it("renders the <Footer /> with populated data", () => {
		const { container, getByText } = render(
			<Footer>
				<Footer.Title>Questions? Contact us</Footer.Title>
				<Footer.Break />
				<Footer.Row>
					<Footer.Column>
						<Footer.Link href="#">About</Footer.Link>
						<Footer.Link href="#">FAQ</Footer.Link>
					</Footer.Column>

					<Footer.Column>
						<Footer.Link href="#">Help</Footer.Link>
						<Footer.Link href="#">Privacy</Footer.Link>
						<Footer.Link href="#">Cookie Preferences</Footer.Link>
					</Footer.Column>

					<Footer.Column>
						<Footer.Link href="#">Refund Policy</Footer.Link>
						<Footer.Link href="#">Terms</Footer.Link>
					</Footer.Column>

					<Footer.Column>
						<Footer.Link href="#">Facebook</Footer.Link>
						<Footer.Link href="#">Twitter</Footer.Link>
					</Footer.Column>
				</Footer.Row>
				<Footer.Break />
				<Footer.Text>Filmix &copy; 2021</Footer.Text>
			</Footer>
		);

		expect(getByText("Questions? Contact us")).toBeTruthy();
		expect(getByText("Cookie Preferences")).toBeTruthy();
		expect(getByText("Terms")).toBeTruthy();
		expect(getByText("Facebook")).toBeTruthy();

		expect(container.firstChild).toMatchSnapshot();
	});
});
