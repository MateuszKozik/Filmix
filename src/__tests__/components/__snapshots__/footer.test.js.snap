// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-sc-10fzgxw-0 gCEKjH"
>
  <p
    class="footer__Title-sc-10fzgxw-4 gerFZx"
  >
    Questions? Contact us
  </p>
  <p
    class="footer__Break-sc-10fzgxw-6 bZbQCC"
  />
  <div
    class="footer__Row-sc-10fzgxw-2 inUtfF"
  >
    <div
      class="footer__Column-sc-10fzgxw-1 cYORVo"
    >
      <a
        class="footer__Link-sc-10fzgxw-3 ePjNiP"
        href="#"
      >
        About
      </a>
      <a
        class="footer__Link-sc-10fzgxw-3 ePjNiP"
        href="#"
      >
        FAQ
      </a>
    </div>
    <div
      class="footer__Column-sc-10fzgxw-1 cYORVo"
    >
      <a
        class="footer__Link-sc-10fzgxw-3 ePjNiP"
        href="#"
      >
        Help
      </a>
      <a
        class="footer__Link-sc-10fzgxw-3 ePjNiP"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-sc-10fzgxw-3 ePjNiP"
        href="#"
      >
        Cookie Preferences
      </a>
    </div>
    <div
      class="footer__Column-sc-10fzgxw-1 cYORVo"
    >
      <a
        class="footer__Link-sc-10fzgxw-3 ePjNiP"
        href="#"
      >
        Refund Policy
      </a>
      <a
        class="footer__Link-sc-10fzgxw-3 ePjNiP"
        href="#"
      >
        Terms
      </a>
    </div>
    <div
      class="footer__Column-sc-10fzgxw-1 cYORVo"
    >
      <a
        class="footer__Link-sc-10fzgxw-3 ePjNiP"
        href="#"
      >
        Facebook
      </a>
      <a
        class="footer__Link-sc-10fzgxw-3 ePjNiP"
        href="#"
      >
        Twitter
      </a>
    </div>
  </div>
  <p
    class="footer__Break-sc-10fzgxw-6 bZbQCC"
  />
  <p
    class="footer__Text-sc-10fzgxw-5 bwPZlU"
  >
    Filmix Â© 2021
  </p>
</div>
`;
